src/components/form/fields/dateField.jsx

```diff
@@ -0,0 +1,170 @@
+                                    required={required}
+                                    onChange={handleDayChange}
+                                    placeholder={i18n('day')}
+                                    options={daysInMonth.map(day => ({ value: day, label: day }))}
+                                    data-testid={`${path}.day`}
+                                    aria-label={i18n('day')}
+                                />
+                            </InputCol>
+                        )
+                    } else if (part === 'month') {
+                        return (
+                            <InputCol key="month" width={50}>
+                                <Select
+                                    name={`${path}.month`}
+                                    value={month || ''}
+                                    hasError={error}
+                                    required={required}
+                                    onChange={handleMonthChange}
+                                    placeholder={i18n('month')}
+                                    options={months.map((month, index) => ({ value: index + 1, label: month }))}
+                                    data-testid={`${path}.month`}
+                                    aria-label={i18n('month')}
+                                />
+                            </InputCol>
+                        )
+                    } else if (part === 'year') {
+                        return (
+                            <InputCol key="year" width={30}>
+                                <Input
+                                    type="number"
+                                    maxlength="4"
+                                    inputmode="numeric"
+                                    name={`${path}.year`}
+                                    value={year || ''}
+                                    hasError={error}
+                                    required={required}
+                                    onChange={handleYearChange}
+                                    placeholder={i18n('year')}
+                                    data-testid={`${path}.year`}
+                                    aria-label={i18n('year')}
+                                />
+                            </InputCol>
+                        )
+                    }
+                })}
+            </InputRow>
+        </FormGroup>
+    )
+}
+
+const dateFormat = locale => DateTime.now().setLocale(locale).toLocaleParts().map(part => {
+    switch (part.type) {
+        case 'day':
+            return 'dd'
+        case 'month':
+            return 'mm';
+        case 'year':
+            return 'yyyy'
+        case 'literal':
+            return part.value;
+    }
+}).join('')
+
+const datetimeValidator = locale => new Validator({
+    rule: (value) => isValued(value) && value.raw.isValid,
+    hint: 'date',
+    parameters: [dateFormat(locale)]
```

Comment:
The current translation for `validation.date` message string is `Cette date n'est pas valide. Le format demandé est mm/jj/aaa`.
I added the right Intl date format in `parameters` to be able to change translation with `Cette date n'est pas valide. Le format demandé est {format}` for example.
But with this new implementation I don't see how a user could fill an invalid date format... :)